index.ts

```diff
@@ -76,23 +99,97 @@ export const sandbox = {
     )[1]
     const client = await _arcFunctions.tables()
     if (seedFile) await seedDb(seedFile, dynamodbClient)
-
-    await Promise.all(
-      // @ts-expect-error table has any type
-      (inv['tables-streams'] ?? []).map(({ table }) =>
-        dynamodbClient.send(
-          new UpdateTableCommand({
-            TableName: client.name(table),
-            StreamSpecification: {
-              StreamEnabled: true,
-              StreamViewType: 'NEW_AND_OLD_IMAGES',
-            },
-          })
+    const tableStreams: TableStreamItem[] = inv['tables-streams']
+    abortController = new AbortController()
+    if (tableStreams?.length) {
+      continueProcessingLoop = true
+      const ddbStreamsClient = new DynamoDBStreamsClient({
+        region: inv.aws.region,
+        endpoint: local.url,
+        credentials,
+      })
+      // Init table streams for those defined
+      await Promise.all(
+        tableStreams
+          .map(({ table }) => table)
+          .filter(
+            (item: string, index: number, tableNames: string[]) =>
+              tableNames.indexOf(item) === index
+          )
+          .map((table: string) =>
+            dynamodbClient.send(
+              new UpdateTableCommand({
+                TableName: client.name(table),
+                StreamSpecification: {
+                  StreamEnabled: true,
+                  StreamViewType: 'NEW_AND_OLD_IMAGES',
+                },
+              })
+            )
+          )
+      )
+      // Reset Stream defaults
+      for (const arcStream of tableStreams) {
+        shardMap[arcStream.table] = []
+        await resetTableStreams(
+          dynamodbClient,
+          ddbStreamsClient,
+          arcStream.table
         )
+      }
+
+      dynamoDbStreamsLoop = periodically(
+        () => streamLoop(ddbStreamsClient),
```

Comment:
Both `ddbStreamsClient` and `streamLoop` are local to this scope, so you don't need to pass `ddbStreamsClient` as an argument.

Reply:
My mistake.